File: libs/config/src/rust/hosts.rs
use envconfig::Envconfig;
use super::init_from_file;

#[derive(Envconfig, Default, Debug)]
pub struct Hosts {
    #[envconfig(from = "GATEWAY")]
    pub gateway: String,

    #[envconfig(from = "AUTH")]
    pub auth: String,

    #[envconfig(from = "OAUTH2")]
    pub oauth2: String,

    #[envconfig(from = "TOKENS")]
    pub tokens: String,

    #[envconfig(from = "PASSPORT")]
    pub passport: String,

    #[envconfig(from = "INTEGRATIONS")]
    pub integrations: String
}

impl Hosts {
    pub fn new() -> Self {
        init_from_file(".env.hosts")
    }
}

File: libs/config/go/hosts.go
package _go

import (
	"fmt"
	"github.com/kelseyhightower/envconfig"
)

type Hosts struct {
	Gateway      string `required:"true"    envconfig:"GATEWAY"`
	TwirpGateway string `required:"true"    envconfig:"TWIRP_GATEWAY"`
	Auth         string `required:"true"    envconfig:"AUTH"`
	Oauth2       string `required:"true"    envconfig:"OAUTH2"`
	Tokens       string `required:"true"    envconfig:"TOKENS"`
	Passport     string `required:"true"    envconfig:"PASSPORT"`
	User         string `required:"true"    envconfig:"USER_RPC"`
	Integrations string `required:"true"    envconfig:"INTEGRATIONS"`
	CdnRpc       string `required:"true"    envconfig:"CDN_RPC"`
	MailerRpc    string `required:"true"    envconfig:"MAILER_RPC"`
}

func NewHosts() *Hosts {
	initEnv()

	var hosts Hosts

	if err := envconfig.Process("", &hosts); err != nil {
		fmt.Printf("failed to process env: %v\n", err)
		panic(err)
	}

	return &hosts
}
